exercises-python/Gui_Pereira/ex02_oo_basics/main.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+import math
+
+class Square:
+    def __init__(self, side1):
+        self.side1 = side1
+
+        self.sides = 2
+        self.area = self.side1**2
+
+    def __str__(self):
+        return f'Square {self.side1}'
+
+class Rectangle:
```

Comment:
You haven't used inheritance here, which is an important feature of Object Oriented code. It's not much of an issue in a simple application such as this, though in more complex cases it can lead to a large amount of code duplication.

I can see you use it in your c++ exercises, so it looks as if this is something you have picked up already!